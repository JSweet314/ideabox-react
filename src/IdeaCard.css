article {
  border-bottom: 1px solid lightgray;
  margin: 1rem 0 0 0;
  width: 50%;
  padding-bottom: 0.5rem;
}

article h2 {
  display: inline-block;
  color: #6d6e71;
  font-family: 'Roboto Slab', serif;
  width: 80%;
  margin: 0;
  font-size: 1.5rem;
}

article p {
  color: #939598;
  width: 100%;
}

.deleteBtn,
.upVoteBtn,
.downVoteBtn {
  float: right;
  border: none;
  margin-left: 0.5rem;
  width: 2rem;
  height: 2rem;
  font-size: 1.5rem;
  cursor: pointer;
  outline: none;
}