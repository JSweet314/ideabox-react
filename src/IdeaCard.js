import React from 'react';
import './styles/IdeaCard.css';
import PropTypes from 'prop-types';

export const IdeaCard = (
  {title, body, id, quality, updateIdeaQuality, removeIdea}
) => {
  return (
    <article>
      <h2>{title}</h2>
      <button
        title="Delete Idea"
        className="deleteBtn"
        onClick={e => removeIdea(id)}
      >
        <span 
          aria-label="delete button"
          role="img" 
          className="deleteBtn"
        >
          ❌
        </span>
      </button>
      <p>{body}</p>
      <p><b>Quality:</b> {quality}
        <button
          title="Downvote Idea"
          className="downVoteBtn"
          onClick={e => updateIdeaQuality(e, id)}
          
        >
          <span 
            aria-label="downvote button" 
            role="img" 
            className="downVoteBtn"
          >
            👎
          </span>
        </button>
        <button
          title="Upvote Idea"
          className="upVoteBtn"
          onClick={e => updateIdeaQuality(e, id)}
          aria-label="upvote button"
        >
          <span 
            aria-label="upvote button" 
            role="img" 
            className="upVoteBtn"
          >
            👍
          </span>
        </button>
      </p>
      
    </article>
  );
};

IdeaCard.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  quality: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  updateIdeaQuality: PropTypes.func.isRequired,
  removeIdea: PropTypes.func.isRequired
};