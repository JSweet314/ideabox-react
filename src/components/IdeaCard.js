import React from 'react';
import PropTypes from 'prop-types';
import '../styles/IdeaCard.css';

const IdeaCard = (
  {title, body, id, quality, updateIdeaQuality, removeIdea}
) => {
  return (
    <article>
      <h2>{title}</h2>
      <button 
        title="Delete Idea"
        className="deleteBtn"
        onClick={() => removeIdea(id)}>
        <span aria-label="delete button" role="img" className="deleteBtn">
          ❌
        </span>
      </button>
      <p>{body}</p>
      <p>
        <button
          title="Upvote Idea"
          className="upVoteBtn"
          onClick={e => updateIdeaQuality(e, id)}>
          <span aria-label="upvote button" role="img" className="upVoteBtn">
            👍
          </span>
        </button>
        <button
          title="Downvote Idea"
          className="downVoteBtn"
          onClick={e => updateIdeaQuality(e, id)}>
          <span aria-label="downvote button" role="img" className="downVoteBtn">
            👎
          </span>
        </button>
        <b>Quality:</b> {quality}
      </p>
    </article>
  );
};

IdeaCard.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  quality: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  updateIdeaQuality: PropTypes.func.isRequired,
  removeIdea: PropTypes.func.isRequired
};

export default IdeaCard;