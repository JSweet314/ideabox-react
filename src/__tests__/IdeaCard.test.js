import React from 'react';
import {mount} from 'enzyme';
import IdeaCard from '../components/IdeaCard.js';

describe('IdeaCard', () => {
  let ideaCard, idea;

  beforeEach(() => {
    idea = {
      id: 1234,
      title: 'Hello',
      body: 'World!',
      quality: 'plausible'
    };

    ideaCard = mount(
      <IdeaCard 
        {...idea}
        updateIdeaQuality={jest.fn()}
        removeIdea={jest.fn()}
      />
    );
  });

  it('should render a card', () => {
    expect(ideaCard).toMatchSnapshot();
  });

  it('should have props title, body, id, quality, updateIdeaQuality, and removeIdea', () => {
    expect(ideaCard.props('title')).toBeDefined();
    expect(ideaCard.props('body')).toBeDefined();
    expect(ideaCard.props('id')).toBeDefined();
    expect(ideaCard.props('quality')).toBeDefined();
    expect(ideaCard.props('removeIdea')).toBeDefined();
    expect(ideaCard.props('updateIdeaQuality')).toBeDefined();
  });

  it('should call update quality if the upvote or down vote buttons are clicked', () => {
    const upVoteBtn = ideaCard.find('button.upVoteBtn');
    const downVoteBtn = ideaCard.find('button.downVoteBtn');

    upVoteBtn.simulate(
      'click',
      ({target: {className: 'upVoteBtn'}}, ideaCard.id)
    );
    upVoteBtn.simulate(
      'click',
      ({ target: { className: 'downVoteBtn' } }, ideaCard.id)
    );
    
    expect(ideaCard.props().updateIdeaQuality).toHaveBeenCalledTimes(2);
  });

  it('should call removeIdea when the delete button is clicked', () => {
    const deleteBtn = ideaCard.find('button.deleteBtn');

    deleteBtn.simulate('click');

    expect(ideaCard.props().removeIdea).toHaveBeenCalledTimes(1);
  });
});
