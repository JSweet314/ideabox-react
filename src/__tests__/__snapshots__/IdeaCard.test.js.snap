// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCard should render a card 1`] = `
<IdeaCard
  body="World!"
  id={1234}
  quality="plausible"
  removeIdea={[Function]}
  title="Hello"
  updateIdeaQuality={[Function]}
>
  <article>
    <h2>
      Hello
    </h2>
    <button
      className="deleteBtn"
      onClick={[Function]}
      title="Delete Idea"
    >
      <span
        aria-label="delete button"
        className="deleteBtn"
        role="img"
      >
        ❌
      </span>
    </button>
    <p>
      World!
    </p>
    <p>
      <b>
        Quality:
      </b>
       
      plausible
      <button
        className="downVoteBtn"
        onClick={[Function]}
        title="Downvote Idea"
      >
        <span
          aria-label="downvote button"
          className="downVoteBtn"
          role="img"
        >
          👎
        </span>
      </button>
      <button
        aria-label="upvote button"
        className="upVoteBtn"
        onClick={[Function]}
        title="Upvote Idea"
      >
        <span
          aria-label="upvote button"
          className="upVoteBtn"
          role="img"
        >
          👍
        </span>
      </button>
    </p>
  </article>
</IdeaCard>
`;
